src/Evaluate.jl

```diff
@@ -611,6 +625,8 @@ function deg2_diff_eval(
     return ResultOk(out, all(isfinite, out))
 end
 
+get_lower_array_type(T,N) = N==1 ? T : AbstractArray{T,N-1}
```

Comment:
Is this used anywhere or could we perhaps delete it?

Reply:
It was used to typeassert the generic eval_tree_array functions. I removed the type assertions so I don't think it's needed.